src/transformState.js

```diff
@@ -5,7 +5,31 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
-}
+  const add = 'addProperties';
+  const remove = 'removeProperties';
+  const clearing = 'clear';
+
+  for (const ch of actions) {
+    if (ch.type === add) {
+      Object.assign(state, ch.extraData);
+    };
+
+    if (ch.type === remove) {
+      for (const ck of ch.keysToRemove) {
+        if (ck in state) {
+          delete state[ck];
+        };
+      };
+    };
+
+    if (ch.type === clearing) {
+      for (const del in state) {
+        delete state[del];
+      };
+    };
```

Comment:
Use [`switch case`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch) in cases where you constantly compare one value with others